tests/utils/test_time.py

```diff
@@ -60,3 +60,47 @@ def test_wait_until(sleep_patch):
+
+
+@pytest.mark.parametrize(
+    ('date_from', 'date_to', 'parts', 'expected'),
+    (
+        (datetime(2019, 12, 12, 0, 1), datetime(2019, 12, 12, 12, 0, 5), 2, '11 hours, 59 minutes'),
+        (datetime(2019, 12, 12, 0, 1), datetime(2019, 12, 12, 12, 0, 5), 1, '11 hours'),
+        (datetime(2019, 12, 12, 0, 1), datetime(2019, 12, 12, 12, 0, 5), None, '11 hours, 59 minutes, 5 seconds'),
+        (datetime(2019, 12, 12, 0, 0), datetime(2019, 12, 11, 23, 59), 2, '1 minute'),
+        (datetime(2019, 11, 23, 20, 9), datetime(2019, 11, 30, 20, 15), 2, '1 week, 6 minutes'),
+        (datetime(2019, 11, 23, 20, 9), datetime(2019, 4, 25, 20, 15), 2, '7 months, 2 weeks'),
+        (datetime(2019, 11, 23, 20, 9), datetime(2019, 4, 25, 20, 15),
+         None, '7 months, 2 weeks, 1 day, 23 hours, 54 minutes'),
+        (datetime(2019, 11, 23, 20, 58), datetime(2019, 11, 23, 21, 3), 2, '5 minutes'),
+        (datetime(2019, 11, 23, 23, 59), datetime(2019, 11, 24, 0, 0), 2, '1 minute'),
+        (datetime(2019, 11, 23, 23, 59), datetime(2022, 11, 23, 23, 0), 2, '3 years, 3 months'),
+        (datetime(2019, 11, 23, 23, 59), datetime(2019, 11, 23, 23, 49, 5), 2, '9 minutes, 55 seconds'),
+    )
+)
+def test_get_duration(date_from: datetime, date_to: datetime, parts: int, expected: str):
+    assert time.get_duration(date_from, date_to, parts) == expected
+
+
+@pytest.mark.parametrize(
+    ('expiry', 'date_from', 'parts', 'expected'),
+    (
+        ('2019-12-12T00:01:00Z', datetime(2019, 12, 12, 12, 0, 5), 2, '2019-12-12 00:01 (11 hours, 59 minutes)'),
+        ('2019-12-12T00:01:00Z', datetime(2019, 12, 12, 12, 0, 5), 1, '2019-12-12 00:01 (11 hours)'),
+        ('2019-12-12T00:01:00Z', datetime(2019, 12, 12, 12, 0, 5),
+         None, '2019-12-12 00:01 (11 hours, 59 minutes, 5 seconds)'),
+        ('2019-12-12T00:00:00Z', datetime(2019, 12, 11, 23, 59), 2, '2019-12-12 00:00 (1 minute)'),
+        ('2019-11-23T20:09:00Z', datetime(2019, 11, 30, 20, 15), 2, '2019-11-23 20:09 (1 week, 6 minutes)'),
+        ('2019-11-23T20:09:00Z', datetime(2019, 4, 25, 20, 15), 2, '2019-11-23 20:09 (7 months, 2 weeks)'),
+        ('2019-11-23T20:09:00Z', datetime(2019, 4, 25, 20, 15), None,
+         '2019-11-23 20:09 (7 months, 2 weeks, 1 day, 23 hours, 54 minutes)'),
+        ('2019-11-23T20:58:00Z', datetime(2019, 11, 23, 21, 3), 2, '2019-11-23 20:58 (5 minutes)'),
+        ('2019-11-23T23:59:00Z', datetime(2019, 11, 24, 0, 0), 2, '2019-11-23 23:59 (1 minute)'),
+        ('2019-11-23T23:59:00Z', datetime(2022, 11, 23, 23, 0), 2, '2019-11-23 23:59 (3 years, 3 months)'),
+        ('2019-11-23T23:59:00Z', datetime(2019, 11, 23, 23, 49, 5), 2, '2019-11-23 23:59 (9 minutes, 55 seconds)'),
+        (None, datetime(2019, 11, 23, 23, 49, 5), 2, None),
+    )
+)
+def test_get_duration_from_expiry(expiry: str, date_from: datetime, parts: int, expected: str):
+    assert time.get_duration_from_expiry(expiry, date_from, parts) == expected
```

Comment:
We don't use `pytest` for our unit tests anymore. This file is an unfortunately leftover, as it was merged during the unittest migration. There's [an issue](https://github.com/python-discord/bot/issues/608) to migrate this file to unittest.

Reply:
Ah I see, I did not realize that, I would like to work on migrating this to `unitest` as well.